.pre-commit-config.yaml

```diff
@@ -0,0 +1,19 @@
+repos:
+    - repo: https://github.com/pre-commit/pre-commit-hooks
+      rev: v4.6.0
+      hooks:
+          - id: check-yaml
+          - id: end-of-file-fixer
+          - id: trailing-whitespace
+          - id: detect-private-key
+    - repo: https://github.com/yoheimuta/protolint
+      rev: v0.50.2 # Select a release here like v0.44.0
+      hooks:
+          - id: protolint
+    - repo: local
+      hooks:
+          - id: cargo fmt
```

Comment:
this job is present in the main workflow
1. means it's done twice
2. in fact, the main job can fail while this one succeeds, because they're not in sync.